src/Elders.Cronus/Projections/Versioning/ProjectionVersionManager.cs

```diff
@@ -57,71 +56,51 @@ public void NotifyHash(string hash)
         {
             EnsureThereIsNoOutdatedBuildingVersions();
 
-            if (HasLiveVersion() == false || IsHashTheLiveOne(hash) == false)
+            if (state.Versions.HasLiveVersion == false || state.Versions.IsHashTheLiveOne(hash) == false)
             {
                 Replay(hash);
             }
         }
 
         public void FinalizeVersionRequest(ProjectionVersion version)
         {
-            var buildingVersion = state.Versions.Where(x => x == version).SingleOrDefault();
-            if (ReferenceEquals(null, buildingVersion) == false)
+            var isVersionFound = state.Versions.Contains(version);
+            if (isVersionFound)
             {
-                var @event = new NewProjectionVersionIsNowLive(state.Id, buildingVersion.WithStatus(ProjectionStatus.Live));
+                var @event = new NewProjectionVersionIsNowLive(state.Id, version.WithStatus(ProjectionStatus.Live));
                 Apply(@event);
             }
 
             EnsureThereIsNoOutdatedBuildingVersions();
         }
 
-        private bool HasLiveVersion() => state.Versions.GetLive() is null == false;
-
-        private bool HasBuildingVersion()
-        {
-            return state.Versions
-                .Where(ver => ver.Status == ProjectionStatus.Building)
-                .OrderByDescending(x => x.Revision)
-                .Any();
-        }
-
         private bool CanReplayHash(string hash)
         {
-            bool isHashTheLiveOne = IsHashTheLiveOne(hash);
+            bool isHashTheLiveOne = state.Versions.IsHashTheLiveOne(hash);
 
-            return HasBuildingVersion() == false && (HasLiveVersion() == false || isHashTheLiveOne);
+            return state.Versions.HasBuildingVersion() == false && (state.Versions.HasLiveVersion == false || isHashTheLiveOne);
```

Comment:
This isn't very easy to read

Reply:
Suggestions?

Reply:
```
bool hasBuildingVersion = state.Versions.HasBuildingVersion();
bool ggwp = hasBuildingVersion == false || isHashTheLiveOne;
return hasBuildingVersion == false && ggwp;
```

Something of the sort, maybe drop the triple hasBuildingVersion check.